# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/ppc-libcompat-syscall.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

libcompat/syscall() support for ppc, tested for both, ppc and ppc64. Needed
(for example) due inotify cruft in latest udev.

  - Rene Rebe <rene@exactcode.de>

--- dietlibc-0.29/libcompat/syscall.S	2005-05-20 16:46:32.000000000 +0200
+++ dietlibc-0.29-fixed/libcompat/syscall.S	2006-01-10 13:58:48.000000000 +0100
@@ -118,8 +118,17 @@
 	nop
 	retl
 	nop
-
+#elif defined (__powerpc__) /* so far works for both, 32/64 bit */
+	mr   %r0,%r3
+	mr   %r3,%r4
+	mr   %r4,%r5
+	mr   %r5,%r6
+	mr   %r6,%r7
+	mr   %r7,%r8
+	mr   %r8,%r9
+	b    __unified_syscall
 #else
 		/* arch not implemented yet */
+#warning "No libcompat syscall() implementation ..."
 #endif
 .size	syscall, . - syscall
